spec/mailers/order_mailer_spec.rb

```diff
@@ -216,4 +216,41 @@
       end
     end
   end
+
+  describe "#invoice_email" do
+    subject(:email) { described_class.invoice_email(order) }
+    let(:order) { create(:completed_order_with_totals) }
+    let!(:invoice_data_generator){ InvoiceDataGenerator.new(order) }
+    let!(:invoice){
+      create(:invoice, order:,
+                       data: invoice_data_generator.serialize_for_invoice)
+    }
+
+    let(:generator){ double(:generator) }
+    let(:renderer){ double(:renderer) }
+    before do
+      allow(OrderInvoiceGenerator).to receive(:new).with(order).and_return(generator)
+      allow(InvoiceRenderer).to receive(:new).and_return(renderer)
+    end
+    context "When invoices feature is not enabled" do
+      it "should call the invoice render with order as argument" do
+        expect(generator).not_to receive(:generate_or_update_latest_invoice)
+        expect(order).not_to receive(:invoices)
+        expect(renderer).to receive(:render_to_string).with(order)
+        email.deliver_now
```

Comment:
I guess this line implicitly tests that the email can be dispatched without raising an error. It would be better to surround it with an expectation (like on line 43-45 above).

`±` I would also `expect`  an email to have been composed with the appropriate info.

There's examples in this file of checking the body of the email. It would be good to check that it has the right PDF attached too, but I'm not sure how to do that.

Reply:
I fixed that on https://github.com/openfoodfoundation/openfoodnetwork/pull/11192/commits/598b75fb696abddf9284f438e522efe1f35936b4